dapp/src/app/components/OpenColony/index.tsx

```diff
@@ -0,0 +1,52 @@
+import * as React from "react";
+
+type Props = {
+  count: number;
+  setAddress: (address: string) => any;
+  getCount: () => any;
+};
+
+export class OpenColony extends React.Component<Props, any> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      address: ""
+    };
+
+    this.handleChange = this.handleChange.bind(this);
```

Comment:
If I understand correctly, binding the 'this' pointer shouldn't be needed in TypeScript, since the handleChange & handleSubmit methods are within a class, the language takes care of that for us.

Reply:
Oh whoops, I just read up and I was wrong. Since this method is a callback from HTML, the binding is necessary for the this pointer to be valid.